_posts/2023-11-10-automated-test-generation-chess-puzzles-with-dafny.markdown

```diff
@@ -0,0 +1,288 @@
+---
+layout: post
+title:  "Automated Test Generation: Chess Puzzles with Dafny"
+date:   2023-11-10 11:00:00 -0500
+author: Sasha Fedchin
+---
+Given all the incredible examples that showcase how powerful Dafny is (did you know that part of Dafny is written in Dafny?), you might think that verification can fully replace testing. And in many cases it does! However if, like a number of industrial Dafny users, you want to integrate Dafny code with existing codebases, you will face challenges that verification alone might not solve:
```

Comment:
> all the incredible examples

which? Be specific.

> (did you know that part of Dafny is written in Dafny?)

Is this one of the incredible examples? The parenthesis and phrasing make it sound like a barely related "by the way"

> you might think

This implies an expected lack of understanding by the reader. Consider only stating the facts.

> And in many cases it does!

I would drop the "And" and the `!`.

> industrial Dafny users

I would refrain from dividing Dafny users into industrial and non-industrial. I suggest dropping "like a number of industrial Dafny users". I think your intention of that subsentence is to state that there what you will describe is a use-case for some people, but I think the reader will mostly care whether this feature matches _their_ use-case.

> you will face challenges that verification alone might not solve

"will" followed by "might" seems strange. Apparently all the challenges you will face might only require verification. Consider changing to "You may face challenges that can not be solved using just verification:"

However, another problem with that sentence is that users will face many challenges, many of which don't relate to correctness at all, like getting the Dafny type checker to pass, learning Dafny, or needing sleep, so saying there are challenges not solved by verification seems like an open door to me.

Reply:
Thank you for raising these issues! I have rewritten the first paragraph to take your suggestions into account, particularly by adding several references to previous blogposts to showcase the power of Dafny and clarify that the challenges I refer to specifically call to testing as an additional quality assurance technique on top of verification.